src/rules/lowercase-name.ts

```diff
@@ -0,0 +1,129 @@
+import { createRule } from './tsUtils';
+import {
+  TSESTree,
+  AST_NODE_TYPES,
+} from '@typescript-eslint/experimental-utils';
+
+const isIdentifier = (
+  callee: TSESTree.LeftHandSideExpression,
+): callee is TSESTree.Identifier => !!callee;
+
+const isItTestOrDescribeFunction = (
+  callee: TSESTree.LeftHandSideExpression,
+): callee is TSESTree.Identifier => {
+  if (!isIdentifier(callee)) {
+    return false;
+  }
+
+  return (
+    callee.name === 'it' || callee.name === 'test' || callee.name === 'describe'
+  );
+};
+
+const isItDescription = (
+  expression?: TSESTree.Expression,
+): expression is TSESTree.Literal | TSESTree.TemplateLiteral =>
+  !!expression &&
+  (expression.type === AST_NODE_TYPES.Literal ||
+    expression.type === AST_NODE_TYPES.TemplateLiteral);
+
+const testDescription = (
+  firstArgument: TSESTree.Literal | TSESTree.TemplateLiteral,
+) => {
+  if (firstArgument.type === AST_NODE_TYPES.Literal) {
+    return firstArgument.value as string;
```

Comment:
Doesn't this hint at an actual error if it's a different kind of literal, e.g. `42`?

Reply:
Hmm, maybe. We don't have a test for that? Probably not.

Reply:
You were right. I've added a test and removed the cast